cmd/quadlet/main.go

```diff
@@ -103,28 +103,47 @@ func Debugf(format string, a ...interface{}) {
 func getUnitDirs(rootless bool) []string {
 	// Allow overriding source dir, this is mainly for the CI tests
 	unitDirsEnv := os.Getenv("QUADLET_UNIT_DIRS")
+	dirs := make([]string, 0)
+
 	if len(unitDirsEnv) > 0 {
-		return strings.Split(unitDirsEnv, ":")
+		for _, eachUnitDir := range strings.Split(unitDirsEnv, ":") {
+			dirs = appendSubPaths(dirs, eachUnitDir)
+		}
+		return dirs
```

Comment:
I can add a check to see if the path provided by this env var is valid (`filepath.Abs(filePath)`) if that sounds good

Reply:
@TomSweeneyRedHat we didn't have that before, so I would not require it in this PR. Moreover, what would you check?

Reply:
FWIW, my main concern was that someone might add something like `.../../../../etc` or some such to get into system or other spaces they should not be.  SO a check to make sure something wasn't a relative path would be a starting point.

Reply:
As @harikannan512 wrote, he can add a check that the path is in fact absolute (the method is `IsAbs`). But, I don't understand how would a relative path be different from an absolute path w.r.t. your concern. AFAIK Quadlet is executed with the user's (root or specific user) permissions. So, Quadlet shouldn't be able to access what the user cannot.

Reply:
From past burns, "shouldn't" doesn't always equal "can't".  I'm not an expert on the how-to's in taking advantage of relative references,  I just know that they have often been manipulated badly and I try to avoid them at all costs.

Reply:
NP, so let's at least enforce that is the environment variable is set, it has to be an absolute path